### PR TITLE
Cleans up ACI code

### PR DESCRIPTION
**Pull request by singhbalwinder**
This PR cleans the ACI code and removes all the unused codes. This makes the logic clear and makes it easy to explore the code.

### CODE REVIEW

**Review by jeff-cohere — APPROVED** [review]
Wow, that's a lot of code removed, and the results are much simpler. Thanks, Balwinder!

**Review by overfelt — APPROVED** [review]

File: components/eam/src/physics/cam/microp_aero.F90 [hunk 1]
```diff
@@ -1,789 +1,686 @@
...
+
+  !! local 
+  integer, parameter :: nbin = 50
+
+  real(r8) :: wlarge,sigma
+  real(r8) :: xx, yy 
+  real(r8) :: zz(nbin) 
+  real(r8) :: wa(nbin) 
+  integer  :: kp(nbin) 
+  integer  :: i, k
```
Comment (brigaudet) [inline components/eam/src/physics/cam/microp_aero.F90]: Maybe just change the above to icol, kk?

**Review by brigaudet — APPROVED** [review]
Wow, a lot shorter, looks great!  Just one minor comment.

File: components/eam/src/physics/cam/macrop_driver.F90 [hunk 2]
```diff
@@ -13,27 +13,14 @@ module macrop_driver
```
Comment (tangshuaiqi) [inline components/eam/src/physics/cam/macrop_driver.F90]: As we are not using macrophysics after introducing CLUBB, Can we get rid of everything and remove macrop_driver?

File: components/eam/src/physics/cam/micro_mg_cam.F90 [hunk 3]
```diff
@@ -677,25 +675,11 @@ subroutine micro_mg_cam_init(pbuf2d)
          call micro_mg_utils_init(r8, rh2o, cpair, tmelt, latvap, latice, &
```
Comment (tangshuaiqi) [inline components/eam/src/physics/cam/micro_mg_cam.F90]: can we also remove other calls in options 0 and 5?

File: components/eam/src/physics/cam/microp_aero.F90 [hunk 4]
```diff
@@ -1,789 +1,686 @@
...
+
+        sigma  = max(0.001_r8, wsig(i,k))
+        wlarge = w0(i,k)
+
+        xx = 6._r8 * sigma / nbin
+
+        do ibin = 1, nbin
+           yy = wlarge - 3._r8*sigma + 0.5*xx
+           yy = yy + (ibin-1)*xx
+           !! wbar = integrator < w * f(w) * dw >
```
Comment (tangshuaiqi) [inline components/eam/src/physics/cam/microp_aero.F90]: this line (and a few others below) may be removed.

**Review by tangshuaiqi — APPROVED** [review]
Great clean up! Just wondering if the entire "macrop_driver" could be removed as macrophysics scheme is not used. and a few other minor questions/comments.

File: components/eam/src/physics/cam/microp_aero.F90 (hunk 1 above)
Comment (singhbalwinder) [reply components/eam/src/physics/cam/microp_aero.F90]: Replaced single letter variable names.

File: components/eam/src/physics/cam/macrop_driver.F90 (hunk 2 above)
Comment (singhbalwinder) [reply components/eam/src/physics/cam/macrop_driver.F90]: Removed macop_driver.F90

File: components/eam/src/physics/cam/micro_mg_cam.F90 (hunk 3 above)
Comment (singhbalwinder) [reply components/eam/src/physics/cam/micro_mg_cam.F90]: Cleaned up the case statements.

File: components/eam/src/physics/cam/microp_aero.F90 (hunk 4 above)
Comment (singhbalwinder) [reply components/eam/src/physics/cam/microp_aero.F90]: This line seems like a comment about the code operation, so I am keeping it for now.

**Review by mingxuanwupnnl — APPROVED** [review]

**Comment by singhbalwinder** [comment]
$ ./cs.status.comp_tmp_aci_clean2_mam4_org_v2_baselines -s
comp_tmp_aci_clean2_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**